src/Compiler/Symbols/FSharpDiagnostic.fs

```diff
@@ -158,21 +158,26 @@ type DiagnosticsScope()  =
             | None -> err ""
 
 /// A diagnostics logger that capture diagnostics, filtering them according to warning levels etc.
-type internal CompilationDiagnosticLogger (debugName: string, options: FSharpDiagnosticOptions) = 
+type internal CompilationDiagnosticLogger (debugName: string, options: FSharpDiagnosticOptions, ?preprocess: (PhasedDiagnostic -> PhasedDiagnostic)) =
     inherit DiagnosticsLogger("CompilationDiagnosticLogger("+debugName+")")
             
     let mutable errorCount = 0
     let diagnostics = ResizeArray<_>()
 
     override _.DiagnosticSink(diagnostic, severity) = 
+        let diagnostic =
+            match preprocess with
+            | Some f -> f diagnostic
+            | _ -> diagnostic
```

Comment:
```suggestion
            | None -> diagnostic
```

Reply:
Hmm... Why? Option type is so ubiquitous that I don't think this affects readability in this case. Do you disagree or are there other considerations?